sale_order_net_price_control/models/sale_order_line.py

```diff
@@ -0,0 +1,53 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import models, _, fields
+from odoo.exceptions import ValidationError
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    offer_price = fields.Boolean()
+    admits_offer_prices = fields.Boolean(related='product_template_id.admits_offer_prices', store=True)
+
+    # FASE 1 desarrollo
+    def _compute_amount(self):
+        super()._compute_amount()
+        # FASE 3 DEL DESARROLLO
+        for record in self:
+            if not self.env.user.has_group('sale_order_net_price_control.group_sale_at_any_price'):
```

Comment:
This condition doesn't depend on `record`, so we call it once outside `for`. If this is `False` we can skip `for` interations, indeed